clients/src/test/java/org/apache/kafka/clients/admin/internals/DescribeTransactionsHandlerTest.java

```diff
@@ -0,0 +1,234 @@
+    }
+
+    private void assertFatalError(
+        DescribeTransactionsHandler handler,
+        String transactionalId,
+        Errors error
+    ) {
+        CoordinatorKey key = coordinatorKey(transactionalId);
+        ApiResult<CoordinatorKey, TransactionDescription> result = handleResponseError(handler, transactionalId, error);
+        assertEquals(emptyList(), result.unmappedKeys);
+        assertEquals(mkSet(key), result.failedKeys.keySet());
+
+        Throwable throwable = result.failedKeys.get(key);
+        assertTrue(error.exception().getClass().isInstance(throwable));
+    }
+
+    private void assertRetriableError(
+        DescribeTransactionsHandler handler,
+        String transactionalId,
+        Errors error
+    ) {
+        ApiResult<CoordinatorKey, TransactionDescription> result = handleResponseError(handler, transactionalId, error);
+        assertEquals(emptyList(), result.unmappedKeys);
+        assertEquals(emptyMap(), result.failedKeys);
+    }
+
+    private void assertUnmappedKey(
+        DescribeTransactionsHandler handler,
+        String transactionalId,
+        Errors error
+    ) {
+        CoordinatorKey key = coordinatorKey(transactionalId);
+        ApiResult<CoordinatorKey, TransactionDescription> result = handleResponseError(handler, transactionalId, error);
+        assertEquals(emptyMap(), result.failedKeys);
+        assertEquals(singletonList(key), result.unmappedKeys);
+    }
+
+    private ApiResult<CoordinatorKey, TransactionDescription> handleResponseError(
+        DescribeTransactionsHandler handler,
+        String transactionalId,
+        Errors error
+    ) {
+        int brokerId = 1;
+
+        CoordinatorKey key = coordinatorKey(transactionalId);
+        Set<CoordinatorKey> keys = mkSet(key);
+
+        DescribeTransactionsResponseData.TransactionState transactionState = new DescribeTransactionsResponseData.TransactionState()
+            .setErrorCode(error.code())
+            .setTransactionalId(transactionalId);
+
+        DescribeTransactionsResponse response = new DescribeTransactionsResponse(new DescribeTransactionsResponseData()
+            .setTransactionStates(singletonList(transactionState)));
+
+        ApiResult<CoordinatorKey, TransactionDescription> result = handler.handleResponse(brokerId, keys, response);
+        assertEquals(emptyMap(), result.completedKeys);
+        return result;
+    }
+
+    private void assertLookup(
+        DescribeTransactionsHandler handler,
+        Set<String> transactionalIds
+    ) {
+        Set<CoordinatorKey> keys = coordinatorKeys(transactionalIds);
+        DescribeTransactionsRequest.Builder request = handler.buildRequest(1, keys);
+        assertEquals(transactionalIds, new HashSet<>(request.data.transactionalIds()));
+    }
+
+    private static CoordinatorKey coordinatorKey(String transactionalId) {
+        return new CoordinatorKey(transactionalId, FindCoordinatorRequest.CoordinatorType.TRANSACTION);
+    }
```

Comment:
nit: We have the same helper methods in few places. I wonder if we could have only one and use it everywhere? `CoordinatorKey` might be a good place (e.g. `CoordinatorKey#transactionalKey`).